### ISSUE TITLE
Make an exception for meaningful punctuations when checking a delimiter run

### ISSUE THREAD

**Issue by dongwoo-kim** [commonmark/commonmark-spec#611]
In [Example 379](https://spec.commonmark.org/0.29/#example-379),

```markdown
a**"foo"**
```

is not strong emphasis, because the opening ** is preceded by an alphanumeric and followed by punctuation, and hence not part of a left-flanking delimiter run.

So the result is : 
```html
<p>a**&quot;foo&quot;**</p>
```

However, punctuation includes some meaningful characters such as `(backtick), < and >. So the opening ** in the following examples are not valid delimiters. 

```markdown
a**`foo`**

a**<span>foo</span>**
```
is converted to :

```html
<p>a**<code>foo</code>**</p>
<p>a**<span>foo</span>**</p>
```

The problem is that without the exception of these meaningful punctuations, there is no way to express these use cases without inserting unnecessary whitespace between ** and preceding alphanumeric. 

```markdown
a **`foo`**

a **<span>foo</span>**
```
is converted to :  

```html
<p>a <strong><code>foo</code></strong></p>
<p>a <strong><span>foo</span></strong></p>
```

Is this intended behavior or something not considered as a specification?

**Comment by jgm** [comment]
I agree that this is an unfortunate limitation.  It would be good to provide some way to express these things.

The best solution, I think, would be provide some way  to manipulate flankingness explicitly, such as the `~` device I suggested in https://talk.commonmark.org/t/beyond-markdown/2787.

This would allow you to do

```
a~**`foo`**~b
```
which would render as
```
a<strong><code>foo</code></strong>b
```

If we had a device like this and required it for all intraword emphasis, it would dramatically simplify the emphasis parsing rules.  But I suppose this is probably a bridge too far.

**Comment by Crissov** [comment]
Well, you could manually insert an HTML coment or an invisible control code or a zero-width character. [Dingus](https://spec.commonmark.org/dingus/?text=a**%22foo%22**%0A%0Aa**%5C%22foo%5C%22**%0A%0Aa**%60foo%60**%0A%0Aa%20**%60foo%60**%0A%0Aa%3C!--%20--%3E**%60foo%60**%0A%0Aa%3C!----%3E**%60foo%60**%0A%0Aa%3C!%3E**%60foo%60**%0A%0Aa%26%2301%3B**%60foo%60**%0A%0Aa%26%23x200D%3B**%60foo%60**%0A%0Aa%26zwj%3B**%60foo%60**%0A%0Aa%26zwnj%3B**%60foo%60**%0A%0Aa%26zwsp%3B**%60foo%60**%0A) 

~~~~ markdown
a**"foo"**

a**\"foo\"**

a**`foo`**

a **`foo`**

a<!-- -->**`foo`**

a<!---->**`foo`**

a<!>**`foo`**

a&#01;**`foo`**

a&#x200D;**`foo`**

a&zwj;**`foo`**

a&zwnj;**`foo`**

a&zwsp;**`foo`**
~~~~

**Comment by jgm** [comment]
That's a good trick. `<??>` also works.

**Comment by dongwoo-kim** [comment]
@Crissov Thanks for the useful information. I hadn't known of these tricks except zero-width space before. For me, HTML comments seem the most natural way. However, as I've been applying this to [the Wysiwyg-markdown convertable editor](https://github.com/nhn/tui.editor), adding unexpected characters into generated markdown text may confuse the users. So, I decided to leave the HTML tag as it is, for example : <code>a\<b>\`foo\`\</b></code>.

@jgm Thanks for the detailed explanation about the spec. As I'm a bit new to the common-mark spec, the link was really helpful to understand the current state and limitation of the spec.

However, I still find it difficult to understand the reason for this limitation. Why `a **<` is a valid left-delimiter-run but `a**<` isn't? Is this based on the heuristic that `**` in the latter case has more chance to be used as normal characters?

**Comment by jgm** [comment]
The basic heuristic is based on a ranking: ALPHANUMERIC > PUNCTUATION > WHITESPACE.
If you go downhill (from greater to less in this ranking), the delimiter can't be left-flanking. (That's what happens with `a**<`, since ALPHANUMERIC > PUNCTUATION.)
If you go uphill, it can't be right-flanking.

**Comment by dongwoo-kim** [comment]
@jgm Thanks for the answer. It helped me to understand the rule more clearly.
